applications/RANSApplication/python_scripts/formulations/turbulence_models/two_equation_turbulence_model_rans_formulation.py

```diff
@@ -0,0 +1,81 @@
+# import kratos
+import KratosMultiphysics as Kratos
+
+# import formulation interface
+from KratosMultiphysics.RANSApplication.formulations.rans_formulation import RansFormulation
+
+# import utilities
+from KratosMultiphysics.RANSApplication import ScalarVariableDifferenceNormCalculationUtility
+from KratosMultiphysics.RANSApplication.formulations.utilities import GetConvergenceInfo
+
+class TwoEquationTurbulenceModelRansFormulation(RansFormulation):
+    def __init__(self, model_part, settings, formulation_1, formulation_2):
+        super().__init__(model_part, settings)
+
+        self.stabilization_method = settings["stabilization_method"].GetString()
+
+        self.formulation_1 = formulation_1
+        self.formulation_1.SetStabilizationMethod(self.stabilization_method)
+        self.AddRansFormulation(self.formulation_1)
+
+        self.formulation_2 = formulation_2
+        self.formulation_2.SetStabilizationMethod(self.stabilization_method)
+        self.AddRansFormulation(self.formulation_2)
+
+        self.echo_level = settings["echo_level"].GetInt()
+        self.nu_t_convergence_utility = ScalarVariableDifferenceNormCalculationUtility(self.GetBaseModelPart(), Kratos.TURBULENT_VISCOSITY)
+        self.SetMaxCouplingIterations(settings["coupling_settings"]["max_iterations"].GetInt())
+
+    def GetMinimumBufferSize(self):
+        if (self.is_steady_simulation):
+            return 1
+        else:
+            return 2
```

Comment:
Shouldn't this value depend on the time scheme considered?

Reply:
Note that you need it to be 3 in the BDF2 case.

Reply:
Each formulation needs to know about their own equation being solved only. So in this case, `two_equation_turbulence_model` only knows about solving solving turbulence model equations only using steady or bossak scheme. So minimum buffer size will be 1 or 2. It does not need to know about how the V-P is solved. The minimum buffer size of V-P will be given by their `RansFormulation`. And the final minimum buffer size is determined by the maximum of the minimum buffer sizes provided by all hierrachical `RansFormulations` using following.

https://github.com/KratosMultiphysics/Kratos/blob/a35bf7bd8cbbaa6603182c1762ac59524520e1b9/applications/RANSApplication/python_scripts/formulations/rans_formulation.py#L157-L167

Reply:
So this is the buffer size for the RANS formulations but no for the VP formulations right? Then what happens if I need a buffer size 3 in my VP solver but 2 for the RANS equations? Is this checked?

Reply:
(the unique way this would work is having different nodes for the VP and the RANS model parts)

Reply:
> So this is the buffer size for the RANS formulations but no for the VP formulations right? Then what happens if I need a buffer size 3 in my VP solver but 2 for the RANS equations? Is this checked?

In this case, the buffer size 3 is chosen, since VP ones need 3 (maximum of minimums). So I don't need to check for the buffer size of RANS, because the buffer size of 2 for RANS will be satisfied by having buffer size of 3. Ideally it would be best to have seperate nodes for VP and RANS model parts, but then I will need lots of copying, or atleast mapping from VP to RANS because Velocity is an input to RANS formulations from VP, Turbulent Viscostiy is an input to VP formulatiosn from RANS as an example. There will be lot of other variables also needs copying in this case such as DISTANCE, NORMALS, etc...

Reply:
OK. Now I understand, as the VP is also a formulation (as the RANS ones) it is automatically checked.

> (the unique way this would work is having different nodes for the VP and the RANS model parts)

This was just a comment. I don't think that separating in two sets of nodes is a good idea in this case.